src/library/objects/ShowcaseExporter.js

```diff
@@ -0,0 +1,873 @@
+            this.canvas.width = this.columnCount * this.rowParams.width;
+        }
+
+        this._fill();
+    };
+
+    ShowcaseExporter.prototype._checkFit = function () {
+        // i don't know how to make this check better
+        // math....
+        var x = 0;
+        var y = 0;
+        for (var type in this.allShipGroups) {
+            if (this.allShipGroups[type].length > 0) {
+                if (y >= this.canvas.height - this.rowParams.height) {
+                    x += this.rowParams.width;
+                    y = 0;
+                }
+                // drawing title
+                y += this.rowParams.height;
+
+                for (var j = 0; j < this.allShipGroups[type].length; j++) {
+                    // drawing ship
+                    y += this.rowParams.height;
+                    if (y >= this.canvas.height) {
+                        x += this.rowParams.width;
+                        y = 0;
+                    }
+                }
+            }
+        }
+
+        return !(x >= this.canvas.width && y !== 0);
+    };
+
+    ShowcaseExporter.prototype._getShips = function () {
+        KC3ShipManager.load();
+        var ships = KC3ShipManager.find(function(s){return s.lock !==0;});
+
+        for (var i in ships) {
+            this.allShipGroups[ships[i].master().api_stype].push(ships[i]);
+            this._shipImages[ships[i].masterId] = null;
+            this.shipCount++;
+        }
+
+        for (i in this.allShipGroups) {
+            if (this.allShipGroups[i].length > 0) {
+                this.allShipGroups[i].sort(function (shipA, shipB) {
+                    if (shipB.level !== shipA.level)
+                        return shipB.level - shipA.level;
+                    else
+                        return shipB.masterId - shipA.masterId;
+                });
+            }
+        }
+    };
+
+    ShowcaseExporter.prototype._drawShip = function (x, y, ship, background) {
+        this.ctx.fillStyle = background;
+        this.ctx.fillRect(x, y, this.rowParams.width, this.rowParams.height);
+        var xOffset = this.rowParams.height / 7;
+
+        this._drawStat(ship.fp, x + xOffset, y, this.rowParams.height / 2, this.rowParams.height / 2, "#ff8888", background);
+        this._drawStat(ship.tp, x + xOffset + this.rowParams.height / 2, y, this.rowParams.height / 2, this.rowParams.height / 2, "#00CCFF", background);
+        this._drawStat(ship.aa, x + xOffset + this.rowParams.height / 2, y + this.rowParams.height / 2, this.rowParams.height / 2, this.rowParams.height / 2, "#ff9900", background);
+        this._drawStat(ship.ar, x + xOffset, y + this.rowParams.height / 2, this.rowParams.height / 2, this.rowParams.height / 2, "#ffcc00", background);
+        this._drawStat(ship.lk, x + xOffset + this.rowParams.height, y, this.rowParams.height / 5, this.rowParams.height / 2, "#66FF66", background);
```

Comment:
@Javran I am not sure that they want see more numbers then it's already have. [From the example](http://i.imgur.com/XIGBXp8.png). But maybe we can add something like vertical progress bar that would represent luck going from min to max instead of just green stripe.